src/openrct2/management/Award.cpp

```diff
@@ -649,13 +665,25 @@ void AwardUpdateAll()
             if (AwardIsDeserved(awardType, activeAwardTypes))
             {
                 // Add award
-                currentAwards.push_back(Award{ 5u, awardType });
-                if (Config::Get().notifications.ParkAward)
-                {
-                    News::AddItemToQueue(News::ItemType::award, AwardNewsStrings[EnumValue(awardType)], 0, {});
-                }
-                windowMgr->InvalidateByClass(WindowClass::ParkInformation);
+                AwardAdd(awardType);
             }
         }
     }
 }
+
+void AwardGrant(AwardType type)
+{
+    auto& currentAwards = getGameState().currentAwards;
+
+    // Remove award type if already granted
+    std::erase_if(currentAwards, [type](const Award& award) { return award.Type == type; });
+
+    // Ensure there is space for the award
+    if (currentAwards.size() >= OpenRCT2::Limits::kMaxAwards)
+    {
+        currentAwards.erase(currentAwards.begin());
+    }
+
+    // Add award
```

Comment:
The comment is superfluous given the name of the function.

Reply:
yea, removed both